marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/auth/SessionMonitorInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package com.marklogic.hub.central.auth;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SessionMonitorInterceptor implements HandlerInterceptor {
+    static final Map<String, Object> stompHeaders = new HashMap<>() {{
+        put("content-type","application/json");
+    }};
+
+    private SimpMessagingTemplate template;
+
+    public SessionMonitorInterceptor(SimpMessagingTemplate template) {
+        this.template = template;
+    }
+
+    @Override
+    public boolean preHandle(
+            HttpServletRequest request, @NotNull HttpServletResponse response, @NotNull Object handler) throws Exception {
+        HttpSession session = request.getSession();
+        if (!(template == null || session == null)) {
```

Comment:
How could the template be null, since it's passed into the constructor? If it could be null, then the constructor should throw an exception because the system is ostensibly misconfigured.